docs/user/azure/install_upi.md

```diff
@@ -0,0 +1,559 @@
+to use one that already exists in your organization:
+
+```sh
+az network dns zone create -g $BASE_DOMAIN_RESOURCE_GROUP -n ${CLUSTER_NAME}.${BASE_DOMAIN}
+```
+
+Create the *private* zone in the same resource group of the rest of this deployment:
+
+```sh
+az network private-dns zone create -g $RESOURCE_GROUP -n ${CLUSTER_NAME}.${BASE_DOMAIN}
+```
+
+## Grant access to the identity
+
+Grant the *Contributor* role to the Azure identity so that the Ingress Operator can create a public IP and its load balancer. You can do that with:
+
+```sh
+export PRINCIPAL_ID=`az identity show -g $RESOURCE_GROUP -n ${RESOURCE_GROUP}-identity --query principalId --out tsv`
+export RESOURCE_GROUP_ID=`az group show -g $RESOURCE_GROUP --query id --out tsv`
+az role assignment create --assignee "$PRINCIPAL_ID" --role 'Contributor' --scope "$RESOURCE_GROUP_ID"
+```
+
+## Deployment
+
+The key part of this UPI deployment are the [Azure Resource Manager][azuretemplates] templates, which are responsible
+for deploying most resources. They're provided as a few json files named following the "NN_name.json" pattern. In the
+next steps we're going to deploy each one of them in order, using [az (Azure CLI)][azurecli] and providing the expected parameters.
+
+## Deploy the Virtual Network
+
+In this example we're going to create a Virtual Network and subnets specifically for the OpenShift cluster. You can skip this step
+if the cluster is going to live in a VNet already existing in your organization, or you can edit the `01_vnet.json` file to your
+own needs (e.g. change the subnets address prefixes in CIDR format).
+
+```sh
+az group deployment create -g $RESOURCE_GROUP \
+  --template-file "01_vnet.json"
+```
+
+Link the VNet just created to the private DNS zone:
+
+```sh
+az network private-dns link vnet create -g $RESOURCE_GROUP -z ${CLUSTER_NAME}.${BASE_DOMAIN} -n ${CLUSTER_NAME}-network-link -v "${RESOURCE_GROUP}-vnet" -e false
+```
+
+## Deploy the image
+
+```sh
+az group deployment create -g $RESOURCE_GROUP \
+  --template-file "02_storage.json" \
+  --parameters vhdBlobURL="${VHD_BLOB_URL}"
+```
+
+## Deploy the load balancers
+
+Deploy the load balancers and public IP addresses:
+
+```sh
+az group deployment create -g $RESOURCE_GROUP \
+  --template-file "03_infra.json"
+```
+
+Create DNS records in the *public* zone for the API public load balancer. Make sure the `BASE_DOMAIN_RESOURCE_GROUP` points to the resource group where the public DNS zone exists.
+
+```sh
+export PUBLIC_IP=`az network public-ip list -g $RESOURCE_GROUP --query "[?name=='${CLUSTER_NAME}-master-pip'] | [0].ipAddress" -o tsv`
+
+az network dns record-set a create -g $BASE_DOMAIN_RESOURCE_GROUP -z ${CLUSTER_NAME}.${BASE_DOMAIN} -n api --ttl 60
+az network dns record-set a add-record -g $BASE_DOMAIN_RESOURCE_GROUP -z ${CLUSTER_NAME}.${BASE_DOMAIN} -n api -a $PUBLIC_IP --ttl 60
```

Comment:
This needs to lose the --ttl 60 (it's an error)

Reply:
Fixed.